Sources/SwiftProtobuf/ProtobufBinaryDecoding.swift

```diff
@@ -682,14 +674,16 @@ private class ProtobufScanner {
 
     // Parse index/type marker that starts each field.
     // This also bookmarks the start of field for a possible skip().
-    func getTagType() throws -> Int? {
+    func getTag() throws -> FieldTag? {
         fieldStartP = p
         fieldStartAvailable = available
         if let t = try getRawVarint() {
             if t > 7 && t < UInt64(UInt32.max) {
```

Comment:
I guess the other option would be to accept an int64 to init and put the check there.